hypothesis-python/tests/array_api/test_arrays.py

```diff
@@ -201,7 +203,12 @@ def test_minimize_float_arrays(xp, xps):
     with other array libraries.
     """
     smallest = minimal(xps.arrays(xp.float32, 50), lambda x: xp.sum(x) >= 1.0)
-    assert xp.sum(smallest) in (1, 50)
+    # TODO_IR the shrinker gets stuck when the first failure is math.inf, because
+    # downcasting inf to a float32 overflows, triggering rejection sampling which
+    # is then immediately not a shrink (specifically it overruns the attempt data).
+    #
+    # this should be resolved by adding float widths to the ir.
+    assert xp.sum(smallest) in (1, 50) or all(math.isinf(v) for v in smallest)
```

Comment:
Ouch!  Makes sense, but it's going to be pretty annoying especially with https://github.com/HypothesisWorks/hypothesis/issues/3959 on the wishlist...

Reply:
If it doesn't turn out to be simple, I'd leave it for later - finishing off the IR migrations is higher priority.